06_predict_survival/nbconverted/plot_survival_pc_comparison.py

```diff
@@ -0,0 +1,459 @@
+axarr[2].set_title('Well-predicted/total ratio vs. PC count')
+
+
+# In[5]:
+
+
+me_performance_df = []
+all_drop_cancer_types = set()
+
+for n_dim in pcs_list:
+    # load results into a single dataframe
+    me_pcs_dir = Path(me_results_dir, 'results_{}_pca'.format(n_dim))
+    me_results_df = su.load_survival_results(me_pcs_dir)
+    me_results_df.rename(columns={'identifier': 'cancer_type',
+                                  'fold_no': 'fold'}, inplace=True)
+    
+    me_count_df = (me_results_df[me_results_df.data_type == 'test']
+        .groupby(['cancer_type', 'training_data'])
+        .count()
+    )
+    problem_df = me_count_df[me_count_df['cindex'] != 16].copy()
+    drop_cancer_types = problem_df.index.get_level_values(0).unique().values
+    
+    me_all_results_df = au.compare_all_data_types(
+        me_results_df[~me_results_df.cancer_type.isin(drop_cancer_types)],
+        SIG_ALPHA,
+        filter_genes=False,
+        compare_ind=True,
+        identifier='cancer_type',
+        metric='cindex')
+    me_all_results_df.rename(columns={'gene': 'cancer_type'}, inplace=True)
+    me_all_results_df['n_dim'] = n_dim
+    me_performance_df.append(me_all_results_df)
+    
+me_performance_df = pd.concat(me_performance_df)
+me_performance_df.training_data.replace(to_replace=training_data_map, inplace=True)
+me_performance_df.head(10)
+
+
+# In[6]:
+
+
+group_cancer_types = me_performance_df.groupby(['cancer_type']).count().n_dim
+max_count = group_cancer_types.max()
+valid_cancer_types = group_cancer_types[group_cancer_types == max_count].index
+print(valid_cancer_types)
+
+
+# In[7]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+sns.set_style('whitegrid')
+
+sns.pointplot(data=me_performance_df, x='n_dim', y='delta_cindex', hue='training_data',
+              hue_order=training_data_map.values())
+plt.xlabel('Number of PCs')
+plt.ylabel('cindex(signal) - cindex(shuffled)')
+plt.title('Performance for varying PC count, averaged over cancer types')
+
+
+# In[8]:
+
+
+cancer_type_avg = (
+    me_performance_df[me_performance_df.cancer_type.isin(valid_cancer_types)]
+      .groupby('cancer_type')
+      .mean()
+).delta_cindex
+cancer_type_avg.sort_values(ascending=False).head(10)
+
+
+# In[9]:
+
+
+cancer_type_sd = me_performance_df.groupby('cancer_type').std().delta_cindex
+cancer_type_cv = cancer_type_avg / cancer_type_sd
+cancer_type_cv.sort_values(ascending=False).head(10)
+
+
+# In[10]:
+
+
+sns.set({'figure.figsize': (7, 6)})
+sns.set_style('whitegrid')
+
+sns.pointplot(data=me_performance_df[me_performance_df.cancer_type == 'pancancer'],
+              x='n_dim', y='delta_cindex', hue='training_data', 
+              hue_order=training_data_map.values())
+plt.xlabel('Number of PCs', size=14)
+plt.ylabel('cindex(signal) - cindex(shuffled)', size=14)
+plt.legend(title='Training data', fontsize=13, title_fontsize=13)
+plt.title('Pan-cancer survival performance, expression/methylation', size=14)
+plt.ylim(-0.05, 0.25)
```

Comment:
Just making sure I'm reading this right, pan-cancer is training models using all cancer types, while the previous figure trains models on each individual type then averages across cancers

Reply:
Yep, exactly right